File: samples/flutter/lib/src/journey_phase.dart
enum JourneyPhase { boarding, running }

JourneyPhase journeyPhase({
  required DateTime scheduledDeparture,
  required int delayMinutes,
  required DateTime now,
}) {
  final delayedDeparture = scheduledDeparture.add(
    Duration(minutes: delayMinutes),
  );
  return now.isBefore(delayedDeparture)
      ? JourneyPhase.boarding
      : JourneyPhase.running;
}

File: samples/ios/Sources/KorailLiveShowcase/JourneyPhase.swift
import Foundation

public enum JourneyPhase: String, Codable, Sendable {
    case boarding
    case running
}

public struct JourneyPhasePolicy: Sendable {
    public init() {}

    public func phase(
        scheduledDeparture: Date,
        delayMinutes: Int,
        now: Date
    ) -> JourneyPhase {
        let delayedDeparture = scheduledDeparture.addingTimeInterval(
            TimeInterval(delayMinutes * 60)
        )
        return now < delayedDeparture ? .boarding : .running
    }
}
